Doc/library/typing.rst

```diff
@@ -582,6 +608,11 @@ These can be used as types in annotations and do not support ``[]``.
       def stop() -> NoReturn:
           raise RuntimeError('no way')
 
+   ``NoReturn`` can also be used as a general
+   `bottom type <https://en.wikipedia.org/wiki/Bottom_type>`_, a type that
+   has no values. The :data:`Never` type provides a more explicit name for
+   the same concept. Type checkers should treat the two equivalently.
```

Comment:
Why not just say that (a) this usage is equivalent to a bottom type, and (b) it is preferred to use `Never` as a bottom type (but only in 3.11+)?

Reply:
Reworded.